chips/rp2040/src/usb.rs

```diff
@@ -0,0 +1,2532 @@
+                        self.handle_epdata_out(12);
+                    }
+                }
+                26 => {
+                    if self.registers.buff_status.is_set(BUFF_STATUS::EP13_IN) {
+                        self.registers
+                            .buff_status
+                            .modify(BUFF_STATUS::EP13_IN::CLEAR);
+                        self.handle_epdata_in(13);
+                    }
+                }
+                27 => {
+                    if self.registers.buff_status.is_set(BUFF_STATUS::EP13_OUT) {
+                        self.registers
+                            .buff_status
+                            .modify(BUFF_STATUS::EP13_OUT::CLEAR);
+                        self.handle_epdata_out(13);
+                    }
+                }
+                28 => {
+                    if self.registers.buff_status.is_set(BUFF_STATUS::EP14_IN) {
+                        self.handle_epdata_in(14);
+                        self.registers
+                            .buff_status
+                            .modify(BUFF_STATUS::EP14_IN::CLEAR);
+                    }
+                }
+                29 => {
+                    if self.registers.buff_status.is_set(BUFF_STATUS::EP14_OUT) {
+                        self.handle_epdata_out(14);
+                        self.registers
+                            .buff_status
+                            .modify(BUFF_STATUS::EP14_OUT::CLEAR);
+                    }
+                }
+                30 => {
+                    if self.registers.buff_status.is_set(BUFF_STATUS::EP15_IN) {
+                        self.handle_epdata_in(15);
+                    }
+                    self.registers
+                        .buff_status
+                        .modify(BUFF_STATUS::EP15_IN::CLEAR);
+                }
+                31 => {
+                    if self.registers.buff_status.is_set(BUFF_STATUS::EP15_OUT) {
+                        self.handle_epdata_out(15);
+                    }
+                    self.registers
+                        .buff_status
+                        .modify(BUFF_STATUS::EP15_OUT::CLEAR);
+                }
+                _ => {}
+            }
+        }
+        self.registers.buff_status.set(0);
+    }
+
+    fn process_requests(&self) {
+        for (endpoint, desc) in self.descriptors.iter().enumerate() {
+            if desc.request_transmit_in.take() {
+                if endpoint == 0 {
+                    self.transmit_in_ep0();
+                } else {
+                    self.transmit_in(endpoint);
+                }
+            }
+            if desc.request_transmit_out.take() {
+                if endpoint == 0 {
+                    self.transmit_out_ep0();
+                } else {
+                    self.transmit_out(endpoint);
+                }
+            }
+        }
+    }
+
+    fn handle_epdata_out(&self, ep: usize) {
```

Comment:
could be worth  returning a result instead of a panic on failure on line 1833 and 1844?

Reply:
We just followed the nrf implementation. I think this modification will require more work and should be in an improvement PR.